trio/_path.py

```diff
@@ -189,75 +294,106 @@ async def open(self, *args, **kwargs):
 
     if TYPE_CHECKING:
         # the dunders listed in _forward_magic that aren't seen otherwise
```

Comment:
It would be nice to get some auto-checking of these signatures - partly to ensure the current signatures are correct, but even more to ensure they're kept up to date with new python versions. Most crude way would be to write a test, another way would be something akin to what I wrote in `_socket`
https://github.com/python-trio/trio/blob/aa32d87dd620511d2f225fe2f7a2f607511636d5/trio/_socket.py#L881-L890
but maybe not if that means this section has to get *even* longer.

Reply:
Unfortunately it's a little tricky, since only some methods are sync vs async, `pathlib.Path` is rewrapped to `trio.Path` in return values, version-only variants, etc.

Reply:
yeah it'd need some wrapping and modifying of the types. But it's not a critical wrapper or anything so we're probably fine even if some mistakes would slip through.